### ISSUE TITLE
Next 12: Build fails when pages nested under `src` directory

### ISSUE THREAD

**Issue by erikshestopal** [vercel/next.js#30399]
### What version of Next.js are you using?

12.0.1

### What version of Node.js are you using?

17.0.0

### What browser are you using?

Chrome

### What operating system are you using?

MacOS

### How are you deploying your application?

Local

### Describe the Bug

Running `next build` when code is nested under a `src` directory causes `next build` to fail at compilation because the `_document` file can't be found. 

### Expected Behavior

Build succeeds. 

### To Reproduce

1. Clone https://github.com/vercel/next-rsc-demo
2. Move `pages` and `components` under a `src` directory
3. Run `next build`

```ts
// .next.config.js

module.exports = {
  experimental: {
    reactRoot: true,
    serverComponents: true,
    concurrentFeatures: true,
    urlImports: ['https://cdn.skypack.dev'],
  },
};

```

```
Module not found: Can't resolve 'code/next-rsc-demo/pages/_document' in 'code/next-rsc-demo/pages/_document'
```

**Comment by shuding** [comment]
This should have been fixed with #30642, please have a try with the latest canary release.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
